### PR TITLE
Fix deprecation warnings

### PR DESCRIPTION
**Pull request by revans2**
This is to fix the deprecation warnings becoming errors from https://github.com/rapidsai/cudf/pull/12548

### CODE REVIEW

**Comment by revans2** [comment]
build

File: tests/src/test/scala/com/nvidia/spark/rapids/ParseDateTimeSuite.scala [hunk 1]
```diff
@@ -279,6 +279,7 @@ class ParseDateTimeSuite extends SparkQueryCompareTestSuite with BeforeAndAfterE
     assert(res)
```
Comment (jlowe) [inline tests/src/test/scala/com/nvidia/spark/rapids/ParseDateTimeSuite.scala:279]: 2023 copyrights

File: tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionTranspilerSuite.scala [hunk 2]
```diff
@@ -880,6 +880,7 @@ class RegularExpressionTranspilerSuite extends FunSuite with Arm {
   }
```
Comment (jlowe) [inline tests/src/test/scala/com/nvidia/spark/rapids/RegularExpressionTranspilerSuite.scala:880]: 2023 copyrights

**Comment by revans2** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**